packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1539,8 +1539,10 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     _history.add(route);
     route.didPush();
     route.didChangeNext(null);
-    if (oldRoute != null)
+    if (oldRoute != null) {
       oldRoute.didChangeNext(route);
+      route.didChangePrevious(oldRoute);
```

Comment:
Is something like this ok for now until your new API or do you prefer changing didPush's signature (breaking API) or add something new?

Reply:
yeah it's fine